### PR TITLE
bug: add maturity gate for zcToken Mint and specs

### PR DESCRIPTION
**Pull request by robrobbins**

### CODE REVIEW

**Review by JTraversa — DISMISSED** [review]
The zcToken looks good, didnt check *every* file but looks like its all a pragma bump

File: test/marketplace/Interfaces.sol [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: UNLICENSED
 
-pragma solidity 0.8.4;
+pragma solidity >= 0.8.4;
```
Comment (sourabhmarathe) [inline test/marketplace/Interfaces.sol:3]: Why aren't we locking in a Solidity version anymore?

File: test/tokens/ZcToken.sol [hunk 2]
```diff
@@ -33,11 +33,13 @@ contract ZcToken is Erc2612, IZcToken {
       return true;
   }
 
+  /// @dev Muturity must not have been reached else we revert here.
   /// @param t Address recieving the minted amount
   /// @param a The amount to mint
   function mint(address t, uint256 a) external onlyAdmin(admin) override returns(bool) {
-      _mint(t, a);
-      return true;
+    require(block.timestamp < maturity, 'maturity reached');
```
Comment (sourabhmarathe) [inline test/tokens/ZcToken.sol:40]: Use a custom error here instead and revert to save gas.

**Review by sourabhmarathe — CHANGES_REQUESTED** [review]
A question and a small gas saving suggestion.

File: test/tokens/ZcToken.sol (hunk 2 above)
Comment (robrobbins) [reply test/tokens/ZcToken.sol:40]: not introducing custom errors in V2.

File: test/marketplace/Interfaces.sol (hunk 1 above)
Comment (robrobbins) [reply test/marketplace/Interfaces.sol:3]: while i prefer locking to a specific version it seems that @JTraversa and everyone else in the space is fine with this. It's all over the v3 repo.

**Review by sourabhmarathe — DISMISSED** [review]

**Pull request merged**